### ISSUE TITLE
Question about the output of NLM

### ISSUE THREAD

**Issue by Bowen-He** [TheAeryan/simple-NLM#1]
Hello, and thank you very much for sharing this user-friendly PyTorch implementation of the neural logic machine!

I was experimenting with the code example provided in the README and noticed that the model can output negative values. I would greatly appreciate it if you could provide some insight into why this occurs, as I was under the impression that the output values should represent the likelihood of concluded predicates and therefore range from 0 to 1.

**Comment by TheAeryan** [comment]
Hi, @Bowen-He 

Thank you very much for your interest! The `NLM` class receives the parameter `activation`, which can be either "sigmoid" or "relu". This activation function is applied to the output predicates of each NLM layer, **except for the final layer**. In the case of the final layer, we only apply a `nn.linear` (or `LazyLinear`) layer, so the outputs can be both positive and negative numbers.

We follow this approach so that the user can then apply any activation function to this output (e.g., softmax).

I hope this answer helps!

**Comment by Bowen-He** [comment]
Thanks so much! That really helps a lot!
